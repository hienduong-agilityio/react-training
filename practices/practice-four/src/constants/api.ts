export const HTTP_METHODS = {
  GET: 'GET',
  POST: 'POST',
  PUT: 'PUT',
  DELETE: 'DELETE'
};

export const ROOT_API = import.meta.env.VITE_API_ENDPOINT;

export const API_URL = {
  PROJECT: `${ROOT_API}/projects`
};
