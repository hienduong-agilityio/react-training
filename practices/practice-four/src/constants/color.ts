// Consider using enums instead of constants
export const COLORS_CLASS = {
  DEFAULT: 'color-default',
  PRIMARY: 'color-primary',
  SUCCESS: 'color-success',
  WARNING: 'color-warning',
  DANGER: 'color-danger '
};
