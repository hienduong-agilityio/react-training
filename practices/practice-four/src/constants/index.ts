// Index constants files
export * from './api';
export * from './color';
export * from './filerSearchOption';
export * from './route';
export * from './rowPerPage';
export * from './store';
export * from './validationMessages';
