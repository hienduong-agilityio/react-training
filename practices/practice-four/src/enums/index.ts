// Index Enum files
export * from './status';
export * from './theme';
