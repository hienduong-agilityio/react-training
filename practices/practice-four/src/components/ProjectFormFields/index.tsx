'use client';

// Enums
import { BUTTON_COLORS, BUTTON_VARIANTS, STATUS } from '@/enums';

// Interfaces
import type { IProjectItemProps } from '@/interfaces';

// Components
import { InputField, SelectField, Spinner, FormActionButton } from '@/components';

// Svg
import arrowRight from '@public/images/arrowRightIcon.svg';

export interface IProjectFormFieldsProps {
  initialFormValues: IProjectItemProps;
  formErrorsMessages: Record<string, string>;
  isQueryProjectDetailPending: boolean;
  title: string;
  handleCancelForm: () => void;
  formAction: (formData: FormData) => void;
}

export const ProjectFormFields = ({
  initialFormValues,
  formErrorsMessages,
  isQueryProjectDetailPending,
  title,
  handleCancelForm,
  formAction
}: IProjectFormFieldsProps): JSX.Element => {
  return (
    <form key={initialFormValues.id} className='relative space-y-4' action={formAction}>
      {isQueryProjectDetailPending && title !== 'Add Project' && (
        <div className='absolute inset-0 flex flex-col gap-5 justify-center items-center bg-opacity-30 bg-white z-10'>
          <Spinner size='large' />
          <p>Loading project details...</p>
        </div>
      )}

      <div className='flex flex-col gap-7 px-4'>
        <InputField
          id='projectName'
          label='Project Name *'
          name='projectName'
          required
          pattern='[a-zA-Z0-9 ]*'
          defaultValue={initialFormValues.projectName}
          errorMessage={formErrorsMessages.projectName}
          customClasses='h-12'
        />

        <InputField
          id='managerName'
          label='Project Manager (PM)'
          name='managerName'
          required
          pattern='[a-zA-Z ]*'
          defaultValue={initialFormValues.manager?.managerName}
          errorMessage={formErrorsMessages['manager.managerName']}
          customClasses='h-12'
        />

        <InputField
          id='managerImage'
          label='Manager Image URL'
          name='managerImage'
          type='url'
          pattern='^(https?:\\/\\/)?([\\da-z\\.-]+)\\.([a-z\\.]{2,6})([\\/\\w \\.-]*)*\\/?$'
          defaultValue={initialFormValues.manager?.managerImage}
          errorMessage={formErrorsMessages['manager.managerImage']}
          customClasses='h-12'
        />

        <SelectField
          id='status'
          label='Status'
          name='status'
          customClasses='text-sm h-12'
          defaultValue={initialFormValues.status}
          errorMessage={formErrorsMessages.status}
          required
        >
          {Object.values(STATUS).map((s) => (
            <option key={s} value={s}>
              {s}
            </option>
          ))}
        </SelectField>

        <InputField
          id='resources'
          label='Resources'
          name='resources'
          type='number'
          defaultValue={initialFormValues.resources}
          errorMessage={formErrorsMessages.resources}
          customClasses='h-12'
        />

        <div className='flex flex-col gap-4 w-1/2'>
          <h4 className='text-lg font-medium text-gray-700'>Project Timeline</h4>
          <div className='flex gap-4'>
            <div className='flex flex-col w-full'>
              <label htmlFor='timeStart'>From</label>
              <InputField
                id='timeStart'
                name='timeStart'
                type='date'
                defaultValue={initialFormValues.timeline?.timeStart}
                customClasses='h-12'
              />
            </div>
            <img src={arrowRight} className='w-5 h-5 mt-6' alt='Arrow Right' />
            <div className='flex flex-col w-full'>
              <label htmlFor='timeEnd'>To</label>
              <InputField
                id='timeEnd'
                name='timeEnd'
                type='date'
                defaultValue={initialFormValues.timeline?.timeEnd}
                customClasses='h-12'
              />
            </div>
          </div>
          {formErrorsMessages['timeline.timeEnd'] && (
            <span className='text-sm text-red-600'>{formErrorsMessages['timeline.timeEnd']}</span>
          )}
        </div>

        <InputField
          id='budget'
          label='Budget'
          name='budget'
          type='number'
          min={0}
          max={1000000}
          required
          defaultValue={initialFormValues.budget}
          errorMessage={formErrorsMessages.budget}
          customClasses='h-12'
        />
      </div>

      <div className='flex justify-end gap-4 bg-white py-5 px-4 rounded-b-xl border-t'>
        <FormActionButton
          type='button'
          color={BUTTON_COLORS.DANGER}
          variant={BUTTON_VARIANTS.OUTLINED}
          onClick={handleCancelForm}
          showSpinner={false}
        >
          Cancel
        </FormActionButton>
        <FormActionButton>{title}</FormActionButton>
      </div>
    </form>
  );
};
