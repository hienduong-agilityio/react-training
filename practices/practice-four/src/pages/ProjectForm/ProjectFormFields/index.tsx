// Libraries
import { FormEvent, useEffect, useState } from 'react';

// Components
import { InputField, SelectField, Button, Spinner } from '@/components';

// Enums
import { BUTTON_COLORS, BUTTON_SIZES, BUTTON_VARIANTS, STATUS } from '@/enums';

// SVG
import arrowRight from '@public/images/arrowRightIcon.svg';

// Types
import type { IProjectItemProps } from '@/interfaces';

/**
* TODO:
  - The form needs to include:
  - Dirty handler: To track and handle changes in the form.
  - Rollback handler: To revert to the previous state if necessary.
  - Temporary save change (draft): To temporarily save changes so users can continue later.
 */

export interface IProjectFormFieldsProps {
  // initialFormValues: The object for form value
  initialFormValues: IProjectItemProps;
  // formErrorsMessages: Error message for form input
  formErrorsMessages: Record<string, string>;
  // isMutating: Mutation pending when edit and add project
  isMutating: boolean;
  // isQueryProjectDetailPending: The pending when get project details
  isQueryProjectDetailPending: boolean;
  // title: The form title
  title: string;
  // handleCancelForm: Handle cancel form button
  handleCancelForm: () => void;
  // handleSubmitForm: Handle submit form button
  handleSubmitForm: (e: FormEvent<HTMLFormElement>) => void;
}

/**
 * ProjectFormFields component
 *
 * @returns {JSX.Element} - ProjectFormFields element.
 */
export const ProjectFormFields = ({
  initialFormValues,
  formErrorsMessages,
  isMutating,
  isQueryProjectDetailPending,
  title,
  handleCancelForm,
  handleSubmitForm
}: IProjectFormFieldsProps): JSX.Element => {
  const [status, setStatus] = useState<string | undefined>(undefined); // Initialize as undefined

  // Set the status when the component mounts or when initialFormValues changes
  useEffect(() => {
    if (initialFormValues.status) {
      setStatus(initialFormValues.status); // Set status from initialFormValues
    }
  }, [initialFormValues.status]);

  const handleStatusChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    setStatus(event.target.value); // Update the local state
  };

  return (
    <form className='relative space-y-4' onSubmit={handleSubmitForm}>
      {isQueryProjectDetailPending && title !== 'Add Project' && (
        <div className='absolute inset-0 flex flex-col gap-5 justify-center items-center bg-opacity-30 bg-white z-10'>
          <Spinner size='large' />
          <p>Please waiting ...</p>
        </div>
      )}
      <div className='flex rounded-t-xl items-center bg-white justify-between p-4'>
        <p className='text-xl font-semibold text-gray-900'>{title}</p>
      </div>
      <div className='flex flex-col gap-7 px-4'>
        <InputField
          id='projectName'
          label='Project Name *'
          errorMessage={formErrorsMessages.projectName}
          customClasses='h-12'
          name='projectName'
          placeholder='Enter project name'
          required
          pattern='[a-zA-Z0-9 ]*'
          defaultValue={initialFormValues.projectName}
        />
        <InputField
          id='managerName'
          label='Project Manager (PM)'
          customClasses='h-12'
          errorMessage={formErrorsMessages['manager.managerName']}
          name='managerName'
          placeholder='Enter project manager'
          required
          pattern='[a-zA-Z ]*'
          defaultValue={initialFormValues.manager?.managerName}
        />
        <InputField
          id='managerImage'
          label='Manager Image URL'
          customClasses='text-sm h-12'
          errorMessage={formErrorsMessages['manager.managerImage']}
          name='managerImage'
          pattern='^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$'
          placeholder='Enter image URL'
          type='url'
          defaultValue={initialFormValues.manager?.managerImage}
        />
        <SelectField
          id='status'
          label='Status'
          errorMessage={formErrorsMessages.status}
          customClasses='text-sm h-12'
          name='status'
          value={status}
          onChange={handleStatusChange}
        >
          <option value={STATUS.ON_TRACK}>{STATUS.ON_TRACK}</option>
          <option value={STATUS.POTENTIAL_RISK}>{STATUS.POTENTIAL_RISK}</option>
          <option value={STATUS.AT_RISK}>{STATUS.AT_RISK}</option>
          <option value={STATUS.ON_HOLD}>{STATUS.ON_HOLD}</option>
        </SelectField>
        <InputField
          id='resources'
          label='Resources'
          customClasses='h-12'
          errorMessage={formErrorsMessages.resources}
          name='resources'
          type='number'
          placeholder='Enter resources'
          defaultValue={initialFormValues.resources}
        />
        <div className='flex flex-col gap-4 w-1/2'>
          <h4 className='text-lg font-medium text-gray-700'>Project Timeline</h4>
          <div className='flex justify-between items-center gap-4'>
            <div className='flex flex-col w-full'>
              <label htmlFor='timeStart' className='mb-2 text-font-medium text-gray-700'>
                From
              </label>
              <InputField
                id='timeStart'
                name='timeStart'
                type='date'
                customClasses='text-sm h-12'
                defaultValue={initialFormValues.timeline?.timeStart}
              />
            </div>
            <img src={arrowRight} className='w-5 h-5 mt-6' alt='Arrow Right' />
            <div className='flex flex-col w-full'>
              <label htmlFor='timeEnd' className='mb-2 text-font-medium text-gray-700'>
                To
              </label>
              <InputField
                id='timeEnd'
                name='timeEnd'
                type='date'
                customClasses='text-sm h-12'
                defaultValue={initialFormValues.timeline?.timeEnd}
              />
            </div>
          </div>
          {formErrorsMessages['timeline.timeEnd'] && (
            <span className='mt-1 text-sm text-red-600'>{formErrorsMessages['timeline.timeEnd']}</span>
          )}
        </div>
        <InputField
          id='budget'
          label='Budget'
          name='budget'
          errorMessage={formErrorsMessages.budget}
          type='number'
          placeholder='US$ 00.00'
          customClasses='text-sm h-12'
          min={0}
          max={1000000}
          required
          defaultValue={initialFormValues.budget}
        />
      </div>
      <div className='flex items-center rounded-b-xl bg-white gap-5 py-5 px-4 justify-end border-gray-200'>
        <Button
          type='button'
          onClick={handleCancelForm}
          variant={BUTTON_VARIANTS.OUTLINED}
          size={BUTTON_SIZES.LARGE}
          color={BUTTON_COLORS.DANGER}
        >
          Cancel
        </Button>
        <Button
          type='submit'
          disabled={isMutating}
          variant={BUTTON_VARIANTS.CONTAINED}
          size={BUTTON_SIZES.LARGE}
          color={BUTTON_COLORS.PRIMARY}
        >
          {isMutating ? <Spinner /> : title}
        </Button>
      </div>
    </form>
  );
};
