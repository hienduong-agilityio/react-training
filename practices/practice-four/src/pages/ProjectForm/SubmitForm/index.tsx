// Library
import { useFormStatus } from 'react-dom';

// Components
import { Button, Spinner } from '@/components';

// Enums
import { BUTTON_COLORS, BUTTON_SIZES, BUTTON_VARIANTS } from '@/enums';

export const SubmitButton = ({ children }: { children: React.ReactNode }) => {
  const { pending } = useFormStatus();

  return (
    <Button
      type='submit'
      disabled={pending}
      variant={BUTTON_VARIANTS.CONTAINED}
      size={BUTTON_SIZES.LARGE}
      color={BUTTON_COLORS.PRIMARY}
    >
      {pending ? <Spinner /> : children}
    </Button>
  );
};
