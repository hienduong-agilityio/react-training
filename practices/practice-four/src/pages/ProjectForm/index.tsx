'use client';

// Libraries
import { useFormState } from 'react-dom';
import { Navigate, useNavigate } from 'react-router-dom';

// Hooks
import { initialProjectFormState, useProjectForm } from '@/hooks';

// Components
import { FormLayout, ProjectFormFields } from '@/components';

const ProjectFormPage = ({ title = '' }: { title: string }) => {
  const navigate = useNavigate();

  const { formAction, initialFormValues, isQueryProjectDetailPending, projectDetailsError } = useProjectForm(title);

  /**
   * useFormState is a hook that allows you to manage the state of a form.
   */
  const [formState, formActionDispatcher] = useFormState(formAction, initialProjectFormState);

  if (projectDetailsError) return <Navigate to='/404' replace />;

  const handleCancelForm = () => {
    if (window.history.length > 1) {
      navigate(-1);
    }
  };
  return (
    <FormLayout title={title}>
      <ProjectFormFields
        initialFormValues={initialFormValues}
        formErrorsMessages={formState.errors}
        isQueryProjectDetailPending={isQueryProjectDetailPending}
        title={title}
        handleCancelForm={handleCancelForm}
        formAction={formActionDispatcher}
      />
    </FormLayout>
  );
};

export default ProjectFormPage;
