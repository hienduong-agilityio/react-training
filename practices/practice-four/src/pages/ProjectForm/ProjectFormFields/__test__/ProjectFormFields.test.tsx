// Libraries
import { render, screen, fireEvent } from '@testing-library/react';

// Pages
import { ProjectFormFields, IProjectFormFieldsProps } from '@/pages/ProjectForm/ProjectFormFields';

// Enums
import { STATUS } from '@/enums';

// Constants
import { VALIDATION_MESSAGES } from '@/constants';

const initialFormValues = {
  id: '123',
  projectName: 'Project A',
  status: STATUS.ON_TRACK,
  manager: {
    managerName: 'John Doe',
    managerImage: 'https://example.jpg'
  },
  resources: '10',
  timeline: {
    timeStart: '2024-01-01',
    timeEnd: '2024-12-31'
  },
  budget: 100000,
  lastUpdate: '2024-10-10'
};

const defaultProps: IProjectFormFieldsProps = {
  initialFormValues,
  formErrorsMessages: {},
  isMutating: false,
  isQueryProjectDetailPending: false,
  title: 'Add Project',
  handleCancelForm: jest.fn(),
  handleSubmitForm: jest.fn()
};

describe('ProjectFormFields', () => {
  beforeAll(() => {
    HTMLFormElement.prototype.requestSubmit = jest.fn();
  });

  const renderComponent = (props: Partial<IProjectFormFieldsProps> = {}) => {
    return render(<ProjectFormFields {...defaultProps} {...props} />);
  };

  it('matches the snapshot for ProjectFormFields', () => {
    const { asFragment } = renderComponent();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the form with initial values', () => {
    renderComponent();

    expect(screen.getByLabelText(/Project Name/i)).toHaveValue('Project A');
    expect(screen.getByLabelText(/Project Manager \(PM\)/i)).toHaveValue('John Doe');
    expect(screen.getByLabelText(/Manager Image URL/i)).toHaveValue('https://example.jpg');
    expect(screen.getByLabelText(/Resources/i)).toHaveValue(10);
    expect(screen.getByLabelText(/Budget/i)).toHaveValue(100000);
    expect(screen.getByLabelText(/From/i)).toHaveValue('2024-01-01');
    expect(screen.getByLabelText(/To/i)).toHaveValue('2024-12-31');
  });

  it('renders default timeline when no timeline data is provided', () => {
    renderComponent({
      initialFormValues: {
        ...initialFormValues,
        timeline: {}
      }
    });

    expect(screen.getByLabelText(/From/i)).toHaveDisplayValue('');
    expect(screen.getByLabelText(/To/i)).toHaveDisplayValue('');
  });

  it('renders a spinner and overlay when isQueryProjectDetailPending is true', () => {
    renderComponent({ isQueryProjectDetailPending: true, title: 'Edit Project' });

    const overlaySpinner = screen.getByText(/please waiting/i);
    expect(overlaySpinner).toBeInTheDocument();
    expect(screen.getByRole('progressbar')).toBeInTheDocument();
  });

  it('calls handleCancelForm when the cancel button is clicked', () => {
    const handleCancelFormMock = jest.fn();
    renderComponent({ handleCancelForm: handleCancelFormMock });

    const cancelButton = screen.getByRole('button', { name: /cancel/i });
    fireEvent.submit(cancelButton);

    expect(handleCancelFormMock).toHaveBeenCalledTimes(0);
  });

  it('calls handleSubmitForm when the form is submitted', () => {
    const handleSubmitFormMock = jest.fn((e) => e.preventDefault());
    renderComponent({ handleSubmitForm: handleSubmitFormMock });

    const submitButton = screen.getByRole('button', { name: /add project/i });
    fireEvent.click(submitButton);

    expect(handleSubmitFormMock).toHaveBeenCalledTimes(1);
  });

  it('displays error messages for invalid form inputs', () => {
    const formErrorsMessages = {
      projectName: VALIDATION_MESSAGES.INVALID_NAME_MESSAGE,
      resources: VALIDATION_MESSAGES.RESOURCES_MIN_VALUE_MESSAGE,
      budget: VALIDATION_MESSAGES.BUDGET_POSITIVE_NUMBER_MESSAGE,
      'timeline.timeEnd': VALIDATION_MESSAGES.TIMELINE_DATE_MESSAGE
    };

    renderComponent({
      formErrorsMessages
    });

    expect(screen.getByText(VALIDATION_MESSAGES.INVALID_NAME_MESSAGE)).toBeInTheDocument();
    expect(screen.getByText(VALIDATION_MESSAGES.RESOURCES_MIN_VALUE_MESSAGE)).toBeInTheDocument();
    expect(screen.getByText(VALIDATION_MESSAGES.BUDGET_POSITIVE_NUMBER_MESSAGE)).toBeInTheDocument();
    expect(screen.getByText(VALIDATION_MESSAGES.TIMELINE_DATE_MESSAGE)).toBeInTheDocument();
  });

  it('renders the correct button text based on mutation state and title', () => {
    renderComponent({ title: 'Edit Project' });
    const submitButton = screen.getByRole('button', { name: /edit project/i });
    expect(submitButton).toBeInTheDocument();

    renderComponent({ isMutating: true });
    const spinnerInButton = screen.getByRole('progressbar');
    expect(spinnerInButton).toBeInTheDocument();
  });
});
