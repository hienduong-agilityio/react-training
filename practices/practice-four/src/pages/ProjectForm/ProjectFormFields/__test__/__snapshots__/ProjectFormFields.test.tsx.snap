// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectFormFields matches the snapshot for ProjectFormFields 1`] = `
<DocumentFragment>
  <form
    class="relative space-y-4"
  >
    <div
      class="flex rounded-t-xl items-center bg-white justify-between p-4"
    >
      <p
        class="text-xl font-semibold text-gray-900"
      >
        Add Project
      </p>
    </div>
    <div
      class="flex flex-col gap-7 px-4"
    >
      <div
        class="flex flex-col"
      >
        <label
          class="mb-2 capitalize text-lg font-medium text-gray-700"
          for="projectName"
        >
          Project Name *
        </label>
        <input
          class="border gap-2 block outline-none focus:ring-4 focus:ring-primary-100 rounded-md py-[6px] px-3 text-gray-900 disabled:text-light disabled:bg-white shadow h-12"
          id="projectName"
          name="projectName"
          pattern="[a-zA-Z0-9 ]*"
          placeholder="Enter project name"
          required=""
          value="Project A"
        />
      </div>
      <div
        class="flex flex-col"
      >
        <label
          class="mb-2 capitalize text-lg font-medium text-gray-700"
          for="managerName"
        >
          Project Manager (PM)
        </label>
        <input
          class="border gap-2 block outline-none focus:ring-4 focus:ring-primary-100 rounded-md py-[6px] px-3 text-gray-900 disabled:text-light disabled:bg-white shadow h-12"
          id="managerName"
          name="managerName"
          pattern="[a-zA-Z ]*"
          placeholder="Enter project manager"
          required=""
          value="John Doe"
        />
      </div>
      <div
        class="flex flex-col"
      >
        <label
          class="mb-2 capitalize text-lg font-medium text-gray-700"
          for="managerImage"
        >
          Manager Image URL
        </label>
        <input
          class="border gap-2 block outline-none focus:ring-4 focus:ring-primary-100 rounded-md py-[6px] px-3 text-gray-900 disabled:text-light disabled:bg-white shadow text-sm h-12"
          id="managerImage"
          name="managerImage"
          pattern="^(https?:\\/\\/)?([\\da-z\\.-]+)\\.([a-z\\.]{2,6})([\\/\\w \\.-]*)*\\/?$"
          placeholder="Enter image URL"
          type="url"
          value="https://example.jpg"
        />
      </div>
      <div
        class="flex flex-col"
      >
        <label
          class="mb-2 capitalize text-lg font-medium text-gray-700"
          for="status"
        >
          Status
        </label>
        <select
          class="border gap-2 block outline-none focus:ring-4 focus:ring-primary-100 rounded-md py-[6px] px-3 text-gray-900 disabled:text-light disabled:bg-white shadow text-sm h-12"
          id="status"
          name="status"
        >
          <option
            value="On track"
          >
            On track
          </option>
          <option
            value="Potential risk"
          >
            Potential risk
          </option>
          <option
            value="At risk"
          >
            At risk
          </option>
          <option
            value="On hold"
          >
            On hold
          </option>
        </select>
      </div>
      <div
        class="flex flex-col"
      >
        <label
          class="mb-2 capitalize text-lg font-medium text-gray-700"
          for="resources"
        >
          Resources
        </label>
        <input
          class="border gap-2 block outline-none focus:ring-4 focus:ring-primary-100 rounded-md py-[6px] px-3 text-gray-900 disabled:text-light disabled:bg-white shadow h-12"
          id="resources"
          name="resources"
          placeholder="Enter resources"
          type="number"
          value="10"
        />
      </div>
      <div
        class="flex flex-col gap-4 w-1/2"
      >
        <h4
          class="text-lg font-medium text-gray-700"
        >
          Project Timeline
        </h4>
        <div
          class="flex justify-between items-center gap-4"
        >
          <div
            class="flex flex-col w-full"
          >
            <label
              class="mb-2 text-font-medium text-gray-700"
              for="timeStart"
            >
              From
            </label>
            <div
              class="flex flex-col"
            >
              <input
                class="border gap-2 block outline-none focus:ring-4 focus:ring-primary-100 rounded-md py-[6px] px-3 text-gray-900 disabled:text-light disabled:bg-white shadow text-sm h-12"
                id="timeStart"
                name="timeStart"
                type="date"
                value="2024-01-01"
              />
            </div>
          </div>
          <img
            alt="Arrow Right"
            class="w-5 h-5 mt-6"
            src=""
          />
          <div
            class="flex flex-col w-full"
          >
            <label
              class="mb-2 text-font-medium text-gray-700"
              for="timeEnd"
            >
              To
            </label>
            <div
              class="flex flex-col"
            >
              <input
                class="border gap-2 block outline-none focus:ring-4 focus:ring-primary-100 rounded-md py-[6px] px-3 text-gray-900 disabled:text-light disabled:bg-white shadow text-sm h-12"
                id="timeEnd"
                name="timeEnd"
                type="date"
                value="2024-12-31"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col"
      >
        <label
          class="mb-2 capitalize text-lg font-medium text-gray-700"
          for="budget"
        >
          Budget
        </label>
        <input
          class="border gap-2 block outline-none focus:ring-4 focus:ring-primary-100 rounded-md py-[6px] px-3 text-gray-900 disabled:text-light disabled:bg-white shadow text-sm h-12"
          id="budget"
          max="1000000"
          min="0"
          name="budget"
          placeholder="US$ 00.00"
          required=""
          type="number"
          value="100000"
        />
      </div>
    </div>
    <div
      class="flex items-center rounded-b-xl bg-white gap-5 py-5 px-4 justify-end border-gray-200"
    >
      <button
        class="flex gap-2 items-center focus:ring-4 focus:ring-primary-100 rounded-md cursor-pointer disabled:opacity-45 bg-white border-2 text-black px-6 py-3 text-lg bg-danger-400 border hover:bg-danger-500 focus:bg-danger-400 hover:text-light focus:text-light"
        type="button"
      >
        Cancel
      </button>
      <button
        class="flex gap-2 items-center focus:ring-4 focus:ring-primary-100 rounded-md cursor-pointer disabled:opacity-45 border-0 text-white px-6 py-3 text-lg bg-primary-500 hover:bg-primary-600 focus:bg-primary-500"
        type="submit"
      >
        Add Project
      </button>
    </div>
  </form>
</DocumentFragment>
`;
